client/core/exchangeratefetcher.go

```diff
@@ -224,6 +227,10 @@ func getInto(ctx context.Context, url string, thing interface{}) error {
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("unexpected response, got status code %d", resp.StatusCode)
```

Comment:
No need to update, but a helpful tool is the `http.StatusText` function.

```go
		return fmt.Errorf("unexpected response, got status code %d (%s)",
			resp.StatusCode, http.StatusText(resp.StatusCode))
```

So you see "429 (Too Many Requests)" or "403 (Forbidden)" e.g.

Reply:
Okay.